### ISSUE TITLE
Missing language code in some translations

### ISSUE THREAD

**Issue by lggruspe** [tatuylonen/wiktextract#242]
I attached a file containing some examples. The most affected languages are Central Kurdish, Northern Kurdish, Sorbian and Sami.

With Central Kurdish, many of the translations put the language code in the "roman" field. There are also some translations that include the language code with the word.

```
659:English good {'lang': 'Sorbian', 'code': None, 'tags': ['Upper-Sorbian', 'masculine'], 'sense': 'healthful', 'word': 'dobry (hsb)', '_dis1': '0 1 0 1 0 0 0 0 1 0 0 1 1 1 1 84 1 1 1 1 1 1 1 1 0 0 0'}
```

[log.txt](https://github.com/tatuylonen/wiktextract/files/11358201/log.txt)

**Comment by kristian-clausal** [comment]
I'm currently kind of taking a look at this (ended up in a rabbit hole about missing senses and broken parsing).

**Comment by kristian-clausal** [comment]
You'll be happy to know that I will soon be looking at the actual issue you highlighted and not the other bugs I found along the way. {{multitrans}} broke with it was on the same line as {{trans-top}} due to the order in which we pre-expand and expand templates, but with a mildly kludgy post_template_fn in a couple of pages the pre-expanded output of multitrans now gets an extra newline at the start of it so the lists that is generated by the template is not broken up.

**Comment by kristian-clausal** [comment]
The missing language codes for the Kurdish languages are caused by old data in a set of script and dialect names, because Central Kurdish has been upgraded to a full language in Wiktionary data [two months ago](https://meta.wikimedia.org/wiki/Requests_for_new_languages/Wiktionary_Central_Kurdish). 

But I did find another way this goes wrong (when removing 'Central Kurdish' from the list of dialects didn't work out immediately): if the name of the language is written different from what we have in our system, like 'Central-Kurdish' instead of 'Central Kurdish', then it breaks down in a similar way. Added some more elif-branches using language name.replace(" ", "-") and vice-versa, which worked out for Central Kurdish and probably others; this is most probably 90% of all cases like this (Fooal Bar vs. Fooal-Bar).

Committed a fix, should drop the number of these kinds of errors dramatically. As a general rule of thumb, we can assume that this kind of error is because we just didn't recognise the language name properly, especially when it's a sublist under another language (most examples I saw were all Kurdish -> Central Kurdish or Sami -> Sami languages in a sublist style examples).

**Comment by kristian-clausal** [comment]
@lggruspe I think I got a lot of these, but if you could regenerate the error data from new json data in case there's anything new that would be nice.

**Comment by kristian-clausal** [comment]
I've committed a bit more code to try to catch different "Northern Sami" style combinations (and fixed a bug I found in my last fix). If you could check out the differences tomorrow (as long as Kaikki updates, that is), that would be grand. I'm only doing incremental changes at a time, but better safe and slow.

**Comment by lggruspe** [comment]
Update: [error data](https://github.com/tatuylonen/wiktextract/files/11429766/missing-code.log)

Most common errors:

```
     25 'lang': 'Note', 'code': None
     24 'lang': 'Sami', 'code': None
     20 'lang': 'Lutshootseed', 'code': None
      9 'lang': '1660, Henry More, An Explanation of the Grand Mystery of Godliness, London', 'code': None
      6 'lang': 'You see, dearest Creature, being now in the pink and prime of health, I could sit up all night', 'code': None
      6 'lang': 'Old Assamese', 'code': None
      5 'lang': 'Mah Meri', 'code': None
      5 'lang': 'Logudorese', 'code': None
      5 'lang': 'Lampung', 'code': None
      5 'lang': 'Jahai', 'code': None
      4 'lang': 'Synonyms', 'code': None
      3 'lang': 'Sorbian', 'code': None
      3 'lang': 'Piemontese', 'code': None
      3 'lang': 'division', 'code': None
      3 'lang': 'Campidanese', 'code': None
      3 'lang': 'Berber', 'code': None
      3 'lang': 'Apachean', 'code': None
      2 'lang': 'Toku-no-Shima', 'code': None
      2 'lang': 'Serawai', 'code': None
      2 'lang': 'See Special', 'code': None
      2 'lang': 'Poitevin-Saintongese', 'code': None
      2 'lang': 'Pesemah', 'code': None
      2 'lang': 'Ngarrindjerri', 'code': None
      2 'lang': 'Mymensinghiya', 'code': None
      2 'lang': 'Murut', 'code': None
      2 'lang': 'multiplication, factorization', 'code': None
      2 'lang': 'Mendriq', 'code': None
      2 'lang': 'Koch Rajbangsi', 'code': None
      2 'lang': 'Idahan', 'code': None
      2 'lang': 'For', 'code': None
      2 'lang': 'Enets', 'code': None
      2 'lang': 'Assyrian Neo Aramaic', 'code': None
      2 'lang': 'Afan Oromo', 'code': None
      2 'lang': 'Afanoromo', 'code': None
      2 'lang': 'addition, summation', 'code': None
      1 'lang': 'West Frysian', 'code': None
      1 'lang': 'Western Parbate Kham', 'code': None
      1 'lang': 'US English', 'code': None
      1 'lang': 'Twi', 'code': None
      1 'lang': 'Tidung', 'code': None
      1 'lang': 'Tacawit', 'code': None
      1 'lang': 'subtraction', 'code': None
      1 'lang': 'Southern Kisi', 'code': None
      1 'lang': 'Soussian', 'code': None
      1 'lang': 'Simalungun', 'code': None
      1 'lang': 'Serudung', 'code': None
      1 'lang': 'Seak', 'code': None
      1 'lang': 'root extraction', 'code': None
      1 'lang': 'Roglai', 'code': None
      1 'lang': 'Otomi', 'code': None
      1 'lang': 'Old kapampangan', 'code': None
      1 'lang': 'Oki-no-Erabu', 'code': None
      1 'lang': 'Nuu-Chah-Nulth', 'code': None
      1 'lang': 'Ngbandi', 'code': None
      1 'lang': 'Miwok', 'code': None
      1 'lang': 'Meriam Mir', 'code': None
      1 'lang': 'Mayan', 'code': None
      1 'lang': 'Manobo', 'code': None
      1 'lang': 'Mandailing', 'code': None
      1 'lang': 'Maguindanaon', 'code': None
      1 'lang': 'Low Sorbian', 'code': None
      1 'lang': 'logarithmization', 'code': None
      1 'lang': 'Lari', 'code': None
      1 'lang': 'Kutai', 'code': None
      1 'lang': 'Kinyarwanda', 'code': None
      1 'lang': 'Kedayan', 'code': None
      1 'lang': 'Karavhay-Balkar', 'code': None
      1 'lang': 'Kamrupia', 'code': None
      1 'lang': 'Jirajira', 'code': None
      1 'lang': 'Iloko', 'code': None
      1 'lang': 'Ìgbò', 'code': None
      1 'lang': 'High Sorbian', 'code': None
      1 'lang': 'Gurindji Kriol', 'code': None
      1 'lang': 'French (Canada)', 'code': None
      1 'lang': 'For quotations using this term, see Citations', 'code': None
      1 'lang': 'exponentiation', 'code': None
      1 'lang': 'Erza', 'code': None
      1 'lang': 'East Malay', 'code': None
      1 'lang': 'Dinajpuria', 'code': None
      1 'lang': 'Church Slavonic', 'code': None
      1 'lang': 'Chinese (Simplified)', 'code': None
      1 'lang': 'Burgundian (Oïl)', 'code': None
      1 'lang': 'Bukar-Sadong', 'code': None
      1 'lang': 'British English', 'code': None
      1 'lang': 'Bilinarra', 'code': None
      1 'lang': 'Assyrian', 'code': None
      1 'lang': 'Afán Oromó', 'code': None
```

**Comment by kristian-clausal** [comment]
"Note" has now its own kludge, the rest seems to be either using terms that aren't classified as single languages in our data (anymore or ever), like the Sami examples (they're just 'Sami'), or typoes or name variants we don't have in our data ("Lutshootseed" for "Lushootseed" and "Old Assamese" are not in the automated language-name data).

I'll take a look at these every now and then, but one-offs like these aren't really worth that much effort and would be better corrected on Wiktionary.

The file containing the language name and code data can be found in wiktextract/wiktextract/data/en/language.json. These are generated from stuff we pull from Wiktionary (see recent commits for this).

**Comment by lggruspe** [comment]
Update: The number of errors is down to 97 as of the most recent dump.

```
en	China	{'lang': 'Assyrian Neo Aramaic', 'code': None, 'sense': 'civilization of the Chinese people', 'roman': 'čīna', 'word': 'ܟ̰ܝܼܢܵܐ', '_dis1': '15 5 5 11 14 10 14 11 11 3'}
en	European bison	{'lang': 'Sami', 'code': None, 'sense': 'the wisent species Bison bonasus', 'word': 'eurohpá bison'}
en	European bison	{'lang': 'Sami', 'code': None, 'sense': 'the wisent species Bison bonasus', 'word': 'visenta'}
en	Faroese	{'lang': 'Sorbian', 'code': None, 'sense': 'language', 'word': 'Ferejšćina', '_dis1': '0 100'}
en	Finnish	{'lang': 'Sorbian', 'code': None, 'sense': 'language', 'word': 'Finšćina'}
en	Roma	{'lang': 'For', 'code': None, 'note': 'not cognates of the Romani term for themselves', 'word': 'translations which are exonyms', '_dis1': '40 30 30'}
en	Roma	{'lang': 'For', 'code': None, 'word': 'see Gypsy.', '_dis1': '40 30 30'}
en	Stanislaus	{'lang': 'Sorbian', 'code': None, 'sense': 'male given name- the standard or most popular form', 'word': 'Stanisław', '_dis1': '94 6'}
en	Swedish	{'lang': 'Sami', 'code': None, 'tags': ['Northern-and-Southern'], 'sense': 'language', 'word': 'ruoŧagiella'}
en	Tonto	{'lang': 'Apachean', 'code': None, 'sense': 'Tonto Apache', 'word': "Dilzhe'e", '_dis1': '51 49'}
en	Tonto	{'lang': 'Apachean', 'code': None, 'sense': 'Tonto Apache', 'word': 'Dilzhe’eh Apache', '_dis1': '51 49'}
en	Tonto	{'lang': 'Apachean', 'code': None, 'sense': 'Tonto Apache', 'word': 'Dilzhę́’é', '_dis1': '51 49'}
en	amen	{'lang': 'Church Slavonic', 'code': None, 'sense': 'so be it', 'roman': 'Aminĭ', 'word': 'Аминь', '_dis1': '0 0'}
en	baby	{'lang': 'Twi', 'code': None, 'sense': 'very young human being', 'word': 'abɔfra', '_dis1': '27 21 17 6 5 3 6 3 4 1 2 0 2 2'}
en	bag	{'lang': 'East Malay', 'code': None, 'sense': 'flexible container', 'word': 'uncaŋ', '_dis1': '29 25 2 2 1 4 1 5 2 2 8 3 2 2 1 11 0 0'}
en	barn owl	{'lang': 'Sami', 'code': None, 'sense': 'Tyto alba', 'word': 'toardnaskuolfi'}
en	beauty	{'lang': 'Berber', 'code': None, 'sense': 'quality of pleasing appearance', 'word': 'aẓli', '_dis1': '72 0 7 3 2 9 2 4 1'}
en	beauty	{'lang': 'Berber', 'code': None, 'sense': 'quality of pleasing appearance', 'word': 'aẓři', '_dis1': '72 0 7 3 2 9 2 4 1'}
en	beauty	{'lang': 'Berber', 'code': None, 'sense': 'quality of pleasing appearance', 'word': 'tiẓilt', '_dis1': '72 0 7 3 2 9 2 4 1'}
en	billabong	{'lang': 'Bilinarra', 'code': None, 'sense': 'Australian backwater lake', 'word': 'langkarna'}
en	billabong	{'lang': 'Gurindji Kriol', 'code': None, 'sense': 'Australian backwater lake', 'word': 'bilibong'}
en	butterfly	{'lang': 'Manobo', 'code': None, 'sense': 'insect', 'word': 'kelivangbang', '_dis1': '100 0 0 0 0 0 0 0 0 0'}
en	butterfly	{'lang': 'Miwok', 'code': None, 'sense': 'insect', 'word': 'kiṭīlah', '_dis1': '100 0 0 0 0 0 0 0 0 0'}
en	corpse	{'lang': 'Western Parbate Kham', 'code': None, 'sense': 'dead body', 'note': 'Taka dialect', 'word': 'मोःरो', '_dis1': '91 9'}
en	cow parsley	{'lang': 'Sami', 'code': None, 'sense': 'a weedy biennial umbellifer', 'word': 'dierpmárássi', '_dis1': '99 1'}
en	cuckoo	{'lang': 'Sami', 'code': None, 'sense': 'the bird', 'word': 'giehka', '_dis1': '28 59 13 0 0 0 0'}
en	daringly	{'lang': 'Synonyms', 'code': None, 'sense': 'In a daring manner', 'word': 'boldly'}
en	daringly	{'lang': 'Synonyms', 'code': None, 'sense': 'In a daring manner', 'word': 'courageously'}
en	daringly	{'lang': 'Synonyms', 'code': None, 'sense': 'In a daring manner', 'word': 'fearlessly'}
en	daringly	{'lang': 'Synonyms', 'code': None, 'sense': 'In a daring manner', 'word': 'impudently'}
en	demerit	{'lang': 'Chinese (Simplified)', 'code': None, 'sense': 'mark for bad conduct', 'roman': 'jìguò', 'word': '记过', '_dis1': '4 96 0'}
en	digging stick	{'lang': 'See Special', 'code': None, 'sense': 'tool used in subsistence agriculture', 'word': 'WhatLinksHere'}
en	digging stick	{'lang': 'See Special', 'code': None, 'sense': 'tool used in subsistence agriculture', 'word': 'digging_stick'}
en	dromedary	{'lang': 'Sami', 'code': None, 'sense': 'single-humped camel', 'word': 'dromedára', '_dis1': '57 43'}
en	ear	{'lang': 'Burgundian (Oïl)', 'code': None, 'sense': 'organ of hearing', 'word': 'oroille', '_dis1': '39 33 0 22 3 1 1 1 1'}
en	ferret	{'lang': 'Sami', 'code': None, 'sense': 'the mammal Mustela putorius furo', 'word': 'albinočáhppesbuoidda', '_dis1': '75 24 1'}
en	ferret	{'lang': 'Sami', 'code': None, 'sense': 'the mammal Mustela putorius furo', 'word': 'ferrehtta', '_dis1': '75 24 1'}
en	five	{'lang': 'Campidanese', 'code': None, 'sense': 'cardinal number', 'word': 'cincu', '_dis1': '90 10'}
en	five	{'lang': 'Logudorese', 'code': None, 'sense': 'cardinal number', 'word': 'chimbe', '_dis1': '90 10'}
en	floater	{'lang': 'British English', 'code': None, 'sense': 'A piece of faeces that floats.', 'word': 'Unsinkable', '_dis1': '2 2 2 77 2 2 1 1 1 1 1 1 1 1 1 1 0 1 0 0 1'}
en	gift	{'lang': 'Logudorese', 'code': None, 'sense': 'something given to another voluntarily, without charge', 'tags': ['masculine'], 'word': 'regalu', '_dis1': '93 1 5 2'}
en	girl	{'lang': 'Koch Rajbangsi', 'code': None, 'sense': 'young female person', 'roman': 'ceṅri', 'word': 'চেংৰী', '_dis1': '20 36 9 14 0 2 2 2 0 1 11 3'}
en	goshawk	{'lang': 'Sami', 'code': None, 'sense': 'bird of prey principally in the genus Accipiter', 'word': 'goahppilfálli'}
en	guinea pig	{'lang': 'Sami', 'code': None, 'sense': 'rodent', 'word': 'mársa', '_dis1': '45 47 3 2 3'}
en	jaguar	{'lang': 'Mayan', 'code': None, 'sense': 'Panthera onca', 'word': "B'alam"}
en	junketing	{'lang': '1660, Henry More, An Explanation of the Grand Mystery of Godliness, London', 'code': None, 'sense': 'A celebratory feast or banquet.', 'note': 'Which things though not so substantially performed'}
en	junketing	{'lang': '1660, Henry More, An Explanation of the Grand Mystery of Godliness, London', 'code': None, 'sense': 'A celebratory feast or banquet.', 'note': 'are notwithstanding in some measure imitated by Witches and Magicians'}
en	junketing	{'lang': '1660, Henry More, An Explanation of the Grand Mystery of Godliness, London', 'code': None, 'sense': 'A celebratory feast or banquet.', 'note': 'that they leave oftentimes the partakers of them as weak and faint almost as if they had eaten nothing […]'}
en	junketing	{'lang': '1660, Henry More, An Explanation of the Grand Mystery of Godliness, London', 'code': None, 'sense': 'A celebratory feast or banquet.', 'note': 'whose viands are observed to afford so little satisfaction to nature'}
en	junketing	{'lang': '1660, Henry More, An Explanation of the Grand Mystery of Godliness, London', 'code': None, 'sense': 'A celebratory feast or banquet.', 'word': 'Book 4'}
en	junketing	{'lang': '1660, Henry More, An Explanation of the Grand Mystery of Godliness, London', 'code': None, 'sense': 'A celebratory feast or banquet.', 'word': 'Chapter 5'}
en	junketing	{'lang': '1660, Henry More, An Explanation of the Grand Mystery of Godliness, London', 'code': None, 'sense': 'A celebratory feast or banquet.', 'word': 'I mean in their junketings'}
en	junketing	{'lang': '1660, Henry More, An Explanation of the Grand Mystery of Godliness, London', 'code': None, 'sense': 'A celebratory feast or banquet.', 'word': 'W. Morden'}
en	junketing	{'lang': '1660, Henry More, An Explanation of the Grand Mystery of Godliness, London', 'code': None, 'sense': 'A celebratory feast or banquet.', 'word': 'p. 110'}
en	junketing	{'lang': 'You see, dearest Creature, being now in the pink and prime of health, I could sit up all night', 'code': None, 'sense': 'A celebratory feast or banquet.', 'note': 'sleep only while the moon covers herself for an instant with a thin veil […]'}
en	junketing	{'lang': 'You see, dearest Creature, being now in the pink and prime of health, I could sit up all night', 'code': None, 'sense': 'A celebratory feast or banquet.', 'word': 'café’s'}
en	junketing	{'lang': 'You see, dearest Creature, being now in the pink and prime of health, I could sit up all night', 'code': None, 'sense': 'A celebratory feast or banquet.', 'word': 'dances'}
en	junketing	{'lang': 'You see, dearest Creature, being now in the pink and prime of health, I could sit up all night', 'code': None, 'sense': 'A celebratory feast or banquet.', 'word': 'junketings: converse for ever'}
en	junketing	{'lang': 'You see, dearest Creature, being now in the pink and prime of health, I could sit up all night', 'code': None, 'sense': 'A celebratory feast or banquet.', 'word': 'plays'}
en	junketing	{'lang': 'You see, dearest Creature, being now in the pink and prime of health, I could sit up all night', 'code': None, 'sense': 'A celebratory feast or banquet.', 'word': 'we might go to moonlight ruins'}
en	lest we forget	{'lang': 'French (Canada)', 'code': None, 'sense': 'Translations', 'word': "n'oublions jamais", '_dis1': '50 50'}
en	light	{'lang': 'Mymensinghiya', 'code': None, 'sense': 'electromagnetic waves', 'roman': 'pośor', 'word': 'পসর', '_dis1': '22 4 2 5 3 3 8 4 11 4 5 5 5 5 4 3 7'}
en	light	{'lang': 'Mymensinghiya', 'code': None, 'sense': 'electromagnetic waves', 'roman': 'pośor', 'word': 'পসৰ', '_dis1': '22 4 2 5 3 3 8 4 11 4 5 5 5 5 4 3 7'}
en	marmot	{'lang': 'Sami', 'code': None, 'sense': 'rodent of the genera Marmota', 'word': 'murmel'}
en	marmot	{'lang': 'Sami', 'code': None, 'sense': 'rodent of the genera Marmota', 'word': 'murmelealli'}
en	minute	{'lang': 'Logudorese', 'code': None, 'sense': 'unit of time', 'tags': ['masculine'], 'word': 'minutu', '_dis1': '48 13 4 0 3 13 0 0 0 0 18'}
en	mother	{'lang': 'Koch Rajbangsi', 'code': None, 'sense': 'female (human) who parents a child, gives birth to a baby, or is pregnant', 'roman': 'maó', 'word': 'মাও', '_dis1': '19 18 18 10 12 1 1 2 0 1 3 1 1 1 9 1 1'}
en	mud	{'lang': 'Kamrupia', 'code': None, 'sense': 'mixture of soil and water', 'roman': 'pek', 'word': 'পেক', '_dis1': '42 34 3 1 2 5 2 0 6 0 4 1'}
en	number	{'lang': 'Soussian', 'code': None, 'sense': 'numeral', 'tags': ['masculine'], 'word': 'nnmra (shi)', '_dis1': '5 33 8 4 8 4 4 12 1 4 0 3 4 2 8 1'}
en	pigeon	{'lang': 'Sami', 'code': None, 'sense': 'bird of the dove and pigeon family Columbidae', 'word': 'dielkoduvvá (palumbus)', '_dis1': '64 4 5 15 5 6'}
en	pigeon	{'lang': 'Sami', 'code': None, 'sense': 'bird of the dove and pigeon family Columbidae', 'word': 'duvvá (livia)', '_dis1': '64 4 5 15 5 6'}
en	premonitory	{'lang': 'For quotations using this term, see Citations', 'code': None, 'sense': 'serving as warning', 'word': 'premonitory.'}
en	rain	{'lang': 'Nuu-Chah-Nulth', 'code': None, 'sense': 'condensed water from a cloud', 'word': 'm̕iƛaa', '_dis1': '93 3 4'}
en	rat	{'lang': 'Sami', 'code': None, 'sense': 'rodent', 'word': 'roahttu', '_dis1': '50 38 1 1 2 1 1 4 0 1'}
en	robin	{'lang': 'Sami', 'code': None, 'sense': 'Erithacus rubecula', 'word': 'guovssoloddi', '_dis1': '80 20 0'}
en	skylark	{'lang': 'Sami', 'code': None, 'sense': 'small brown passerine bird', 'word': 'leivvoš'}
en	state	{'lang': 'Assyrian Neo-Aramiac', 'code': None, 'sense': 'computing: the stable condition of a processor during a particular clock cycle', 'roman': 'qyāmā', 'tags': ['masculine'], 'word': 'ܩܝܵܡܵܐ', '_dis1': '7 5 23 10 10 6 5 5 1 8 3 1 1 4 1 2 1 1 2 3'}
en	sun	{'lang': 'Old kapampangan', 'code': None, 'sense': 'the star around which the Earth revolves', 'word': 'ari'}
en	swallow	{'lang': 'Sami', 'code': None, 'sense': 'bird', 'word': 'láhtospálfu'}
en	swallow	{'lang': 'Sami', 'code': None, 'sense': 'bird', 'word': 'spálfu'}
en	swallow	{'lang': 'Sami', 'code': None, 'sense': 'bird', 'word': 'suorrespálfu'}
en	three	{'lang': 'Campidanese', 'code': None, 'sense': 'cardinal number 3', 'word': 'tresi', '_dis1': '92 8'}
en	three	{'lang': 'Logudorese', 'code': None, 'sense': 'cardinal number 3', 'word': 'tres', '_dis1': '92 8'}
en	thrush nightingale	{'lang': 'Sami', 'code': None, 'sense': 'Luscinia luscinia', 'word': 'čuohtegielaš'}
en	tiger	{'lang': 'Kedayan', 'code': None, 'sense': 'The mammal Panthera tigris', 'word': 'haimau', '_dis1': '32 31 31 0 1 0 0 0 0 0 0 2 0 1 1'}
en	tiger	{'lang': 'Pesemah', 'code': None, 'sense': 'The mammal Panthera tigris', 'word': "ne'ngau", '_dis1': '32 31 31 0 1 0 0 0 0 0 0 2 0 1 1'}
en	tiger	{'lang': 'Pesemah', 'code': None, 'sense': 'The mammal Panthera tigris', 'word': 'setue', '_dis1': '32 31 31 0 1 0 0 0 0 0 0 2 0 1 1'}
en	tiger	{'lang': 'Serawai', 'code': None, 'sense': 'The mammal Panthera tigris', 'word': 'setuau', '_dis1': '32 31 31 0 1 0 0 0 0 0 0 2 0 1 1'}
en	tiger	{'lang': 'Serawai', 'code': None, 'sense': 'The mammal Panthera tigris', 'word': 'setuo', '_dis1': '32 31 31 0 1 0 0 0 0 0 0 2 0 1 1'}
en	turtle dove	{'lang': 'Sami', 'code': None, 'sense': 'bird in the genus Streptopelia', 'word': 'turtelduvvá'}
en	two	{'lang': 'Campidanese', 'code': None, 'sense': 'numerical value', 'word': 'duus', '_dis1': '100 0'}
en	two	{'lang': 'Logudorese', 'code': None, 'sense': 'numerical value', 'word': 'duos', '_dis1': '100 0'}
en	unsinkable	{'lang': 'US English', 'code': None, 'sense': 'A piece of faeces that floats.', 'word': 'floater', '_dis1': '1 1 99'}
en	water	{'lang': 'Ngbandi', 'code': None, 'sense': 'liquid H₂O', 'word': 'ngú', '_dis1': '31 24 24 4 0 0 2 1 1 3 4 1 1 1 1 1 2'}
en	water	{'lang': 'Otomi', 'code': None, 'tags': ['México Otomi'], 'sense': 'liquid H₂O', 'word': 'deje', '_dis1': '31 24 24 4 0 0 2 1 1 3 4 1 1 1 1 1 2'}
en	wild boar	{'lang': 'Sami', 'code': None, 'sense': 'Sus scrofa', 'word': 'meahccespiidni'}
en	winter	{'lang': 'Dinajpuria', 'code': None, 'sense': 'fourth season, marked by short days and the lowest temperatures', 'roman': 'jaṛ', 'word': 'জাড়', '_dis1': '65 5 3 6 20'}
```

**Comment by kristian-clausal** [comment]
Some of those are just Wiktionary errors: daringly has a copy pasted synonym gloss that I've already moved, `digging stick` is a mess that someone should take a look at, Sorbian, Sami and some others are not classified as single languages anymore, some of these are typoes and some are just language names not found in the list, like Dinajpuria.

`Assyrian Neo-Aramaic` is frustrating because it mixes spaces and dashes, so fixing "Assyrian Neo Aramaic" with "Assyrian-Neo-Aramaic" doesn't work, like we do now. Generating every single possible permutation of dashes and spaces in a word is overkill for this; we're down to sub-100 errors (most of which are not bugs but bad source data in) so I'm not going to add it. It can be most easily fixed on Wiktionary.

tldr; most of this is correct because there should not be a language code for these entries, either because it's a language name without a wiktionary language code attached to it, or it's an error on the Wiktionary side of things. This being the case, I think it's time to close this issue as pretty much solved.
